lib/active_reporting/database_adapters/mysql_adapter.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module ActiveReporting
+  module DatabaseAdapters
+    # Database adapters are here to solve SQL problems in the
+    # most idiomatic way in each database
+    class MysqlAdapter < Base
+      # Generate SQL snippet with DATE_TRUNC
+      # @param [String] interval
+      # @param [String] field
+      # @return [String]
+      def date_trunc(interval, field)
+        clean_sql(
+          <<-SQL
+            DATE_ADD(
+              "#{super_old_base_date}",
+              INTERVAL TIMESTAMPDIFF(
+                #{interval.upcase},
+                "#{super_old_base_date}",
+                #{field}
+              ) #{interval.upcase}
+            )
+          SQL
+        )
+      end
```

Comment:
I've been playing around with different variants of replicating `date_trunc`, including this one. This approach is about 2x faster according to my primitive metrics, but it falls short in truly replicating `date_trunc`, mostly patently shown by the limit on supported datetime_hierarchies.

Take a look at [this code](https://github.com/LawKick/active_reporting/blob/06a5228728c99d18c4be252362667c8108734af2/lib/active_reporting/function_adapters/mysql.rb#L49) and consider if it would be useful. It is ugly but may provide a better parallel in functionality.